public/app/features/panel/panellinks/link_srv.ts

```diff
@@ -209,7 +211,7 @@ export class LinkSrv implements LinkService {
         value: variablesQuery,
       },
     });
-
+    info.href = getConfig().disableSanitizeHtml ? info.href : sanitizeUrl(info.href);
```

Comment:
The only problem I have with this approach is that `info.href` will not only be sanitized against XSS-type urls, but it will also be html-escaped.  The problem with that is that when we now put that value into html, we have to be aware that it is already escaped, and not escape it again (It will contain `&lt;` in place of `<` and if we escape that again we'd end up with `&amp;lt;`).

Reply:
I think we can make this assumption as consumers of this data links model should not care about any escaping IMO. One problematic place could be the panel links corner where we actually do the escaping. I'm gonna take a look at that. 

But I'd rather have both sanitisation and escaping here  than scattering this logic across all of the components that use this model.

Reply:
I agree, the best solution would be for the service to handle sanitization so it gives a "trusted" url, but let the html-escaping happen when actually building the html so that it's clear what's going on and so that we don't have issues eg with react escaping the values when embedded in jsx